pages/signup.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import Input from '../components/Input';
+import Button from '../components/Button';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useForm, FormProvider } from 'react-hook-form';
+import axios from '../config/axios';
+
+type FormData = {
+  name: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  b_date: string;
+};
+
+const schema = yup.object().shape({
+  name: yup.string().required(),
+  email: yup.string().email().required(),
+  password: yup.string().min(8).max(20).required(),
+  confirmPassword: yup.string().min(8).max(20).required(),
+  b_date: yup.string().required(),
+});
+
+type Props = {
+  toggleShow: boolean;
+  setToggleShow: (toggleShow: boolean) => void;
+};
+
+function Signup({ toggleShow, setToggleShow }: Props) {
+  const methods = useForm<FormData>({
+    resolver: yupResolver(schema),
+  });
+  const onSubmit = methods.handleSubmit((data: FormData) => {
+    const { name, email, password, b_date } = data;
+
+    axios
+      .post('/auth/register', {
+        name,
+        email,
+        password,
+        b_date,
+      })
+      .then((res) => {
+        console.log(res);
```

Comment:
Give feedback to the user when the data is send, whether there are errors or the request was succesful